### PR TITLE
rafactor base integration tests

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
- Prevent application context from spinning up multiple times by moving all @MockBean and @SpyBean to BaseIntegrationTest

- Add -Dspring.test.context.cache.maxSize=1. See:
https://stackoverflow.com/questions/54117965/spring-integration-test-consumes-a-lot-of-memory-uses-a-high-number-of-duplicat

- Refactor BaseIntegrationTests:
- - Remove all unnecessary @MockBeans and @SpyBeans
- - Move all @_Inject and @_Autowired out of BaseIntegrationTest

### CODE REVIEW

File: besluit/src/test/kotlin/com/ritense/besluit/BaseIntegrationTest.kt [hunk 1]
```diff
@@ -78,9 +73,6 @@ class BaseIntegrationTest : BaseTest() {
     @Autowired
     lateinit var connectorTypeInstanceRepository: ConnectorTypeInstanceRepository
 
-    @MockBean
-    lateinit var mailSender: MailSender
```
Comment (theo-ritense) [inline besluit/src/test/kotlin/com/ritense/besluit/BaseIntegrationTest.kt:82]: While the bean was not used in the code, the mock prevents the application from using a real mailsender during tests.

File: core/src/test/java/com/ritense/valtimo/BaseIntegrationTest.java [hunk 2]
```diff
@@ -16,51 +16,24 @@
...
-    public RuntimeService runtimeService;
-
     @Inject
     public ContextService contextService;
 
-    @MockBean
-    public UserManagementService userManagementService;
-
-    @MockBean
-    public MailSender mailSender;
```
Comment (theo-ritense) [inline core/src/test/java/com/ritense/valtimo/BaseIntegrationTest.java:50]: Same as previous comment. Are we sure the test do/can not use a real MailSender if you remove this mock?

File: document-generation/smartdocuments/src/test/kotlin/com/ritense/smartdocuments/BaseIntegrationTest.kt [hunk 3]
```diff
@@ -16,35 +16,24 @@
...
-    lateinit var userManagementService: UserManagementService
-
-    @MockBean
-    lateinit var currentUserService: CurrentUserService
-
     @MockBean
     lateinit var resourceService: ResourceService
 
-    @MockBean
-    lateinit var mailSender: MailSender
```
Comment (theo-ritense) [inline document-generation/smartdocuments/src/test/kotlin/com/ritense/smartdocuments/BaseIntegrationTest.kt:49]: same

File: form-flow-valtimo/src/test/kotlin/com/ritense/valtimo/formflow/BaseIntegrationTest.kt [hunk 4]
```diff
@@ -16,27 +16,12 @@
...
-import org.springframework.boot.test.mock.mockito.MockBean
 import org.springframework.test.context.junit.jupiter.SpringExtension
 
 @SpringBootTest
 @ExtendWith(value = [SpringExtension::class])
 @Tag("integration")
-class BaseIntegrationTest {
-
-    @MockBean
-    lateinit var mailSender: MailSender
```
Comment (theo-ritense) [inline form-flow-valtimo/src/test/kotlin/com/ritense/valtimo/formflow/BaseIntegrationTest.kt:34]: same

File: form/src/test/java/com/ritense/form/BaseIntegrationTest.java [hunk 5]
```diff
@@ -16,51 +16,13 @@
...
 @SpringBootTest
 @ExtendWith(SpringExtension.class)
 @Tag("integration")
 public abstract class BaseIntegrationTest extends BaseTest {
-
-    @MockBean
-    public UserManagementService userManagementService;
-
-    @MockBean
-    public MailSender mailSender;
```
Comment (theo-ritense) [inline form/src/test/java/com/ritense/form/BaseIntegrationTest.java:43]: same

File: form/src/test/java/com/ritense/form/FormTestConfiguration.java [hunk 6]
```diff
@@ -31,16 +27,6 @@ public static void main(String[] args) {
         SpringApplication.run(FormTestConfiguration.class, args);
     }
 
-    @Bean
-    public ResourceService resourceService() {
-        return mock(ResourceService.class);
-    }
-
-    @Bean
-    public MailSender mailSender() {
```
Comment (theo-ritense) [inline form/src/test/java/com/ritense/form/FormTestConfiguration.java:40]: same

File: haalcentraal/haalcentraal-brp/src/test/kotlin/com/ritense/haalcentraal/brp/BaseIntegrationTest.kt [hunk 7]
```diff
@@ -59,12 +56,6 @@ class BaseIntegrationTest : BaseTest() {
     @Autowired
     lateinit var connectorDeploymentService: ConnectorDeploymentService
 
-    @MockBean
-    lateinit var userManagementService: UserManagementService
-
-    @MockBean
```
Comment (theo-ritense) [inline haalcentraal/haalcentraal-brp/src/test/kotlin/com/ritense/haalcentraal/brp/BaseIntegrationTest.kt:65]: I think you get the drift from here...

File: besluit/src/test/kotlin/com/ritense/besluit/BaseIntegrationTest.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply besluit/src/test/kotlin/com/ritense/besluit/BaseIntegrationTest.kt:82]: I replaced the MailSender MockBean with this dependency:
```testImplementation project(":mail:local-mail")```
Which does nothing except for logging

**Review by theo-ritense — APPROVED** [review]

**Pull request closed without merging**